config.c

```diff
@@ -1097,7 +1088,17 @@ int git_config_bool_or_int(const char *name, const char *value, int *is_bool)
 		return v;
 	}
 	*is_bool = 0;
-	return git_config_int(name, value);
+	if (git_parse_int(value, &v))
+		return v;
+
+	if (!strcmp(name, "GIT_TEST_GETTEXT_POISON"))
+		/*
+		 * We explicitly *don't* use _() here since it would
+		 * cause an infinite loop with _() needing to call
+		 * use_gettext_poison().
+		 */
+		die("bad boolean config value '%s' for '%s'", value, name);
+	die(_("bad boolean config value '%s' for '%s'"), value, name);
```

Comment:
It is really hard to spot the difference between these two `die()` calls without further guidance. The difference, of course, is that the `GIT_TEST_GETTEXT_POISON` variant does _not_ use translation, while the regular code path _does_.

Maybe there is a way to make this more obvious, e.g. by adding another comment, or using `else` here?